tests/suites/static_analysis/lint_go.sh

```diff
@@ -42,8 +42,33 @@ run_go_tidy() {
 	fi
 }
 
+join () {
+	local IFS="$1"
+	shift
+	echo "$*"
+}
+
 run_govulncheck() {
-	govulncheck "github.com/juju/juju/..."
+	ignore=(
+		# false positive vulnerability in github.com/canonical/lxd. This is resolved in lxd-5.21.2.
+		# Anyway, it does not affect as as we only use client-side lxc code, but the vulnerability is
+		# server-side.
+		# https://pkg.go.dev/vuln/GO-2024-3312
+		# https://pkg.go.dev/vuln/GO-2024-3313
+		"GO-2024-3312"
+		"GO-2024-3313"
```

Comment:
I personally don't like this from a security point of view. Yes, I know the code isn't executed, but the solution is just to update the dependency.

Reply:
Also, govulncheck states the following:

> Govulncheck is a low-noise tool that helps you find and fix vulnerable dependencies in your Go projects. It does this by scanning your project’s dependencies for known vulnerabilities and then identifying any direct or indirect calls to those vulnerabilities in your code.

So I'd be interested to know if the tool is either incorrect (false positive) or we are linking to it.

 - https://go.dev/doc/tutorial/govulncheck

Reply:
Long-term, yes. However `govulncheck` will still report this as a vulnerability even when we have bumped the version.

We need to wait for the vulndb team to update their db before your fix will unblock static analysis

https://github.com/golang/vulndb/issues/3317

Reply:
This seems to be the fix https://github.com/canonical/lxd/pull/14149/files#diff-a07ebbb9ab09ffec5606ca486752c5ac2b48722626b8c5efa241e14060a05ca8
=